students/tammyd/lesson05/mailroom_pt3.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python
+
+"""
+
+Mailroom program for a non profit
+
+"""
+
+import sys
+
+donor_db = {"britney spears": ("Britney Spears", [50.99, 100.99, 1000.99]),
+            "christina aguilera": ("Christina Aguilera", [50.99, 100.99, 2000.99]),
+            "mandy moore": ("Mandy Moore", [50.99, 100.99, 3000.99]),
+            "jessica simpson": ("Jessica Simpson", [50.99, 100.99, 4000.99]),
+            "queen bey": ("Queen Bey", [50.99, 100.99, 5000.99]),
+            }
+
+def donor_database():
+    return donor_db
+
+
+def mainloop():
+    print("Welcome to the Mailroom \n")
+    main_menu()
+
+
+def main_menu():
+    print("Main Menu:")
+    print(
+      "T - Send a Thank You\n"
+      "L - Send a Letter to all Donors\n"
+      "R - Create a Report\n"
+      "Q - Quit the Program"
+      )
+
+    menu_dict = {"T": thank_you, 
+                 "L": letters_to_everyone, 
+                 "R": create_report, 
+                 "Q": quit_program
+                 }
+
+    while True:
+        menu_selection = input("\nChoose a letter:  ")
+        menu_selection = menu_selection.upper().strip()
+        try:
+            #can you remind me how the () here works?
+            return menu_dict[menu_selection]()
+
+            # if menu_selection == "T":
+            #     thank_you()
+            # elif menu_selection == "L":
+            #     letters_to_everyone()
+            # elif menu_selection == "R":
+            #     create_report()
+            # elif menu_selection == "Q":
+            #     quit_program()
+
+        #why is my exception not being raised here when I make an invalid entry?
+        except (KeyError):
```

Comment:
So you're catching the keyerrors so the exception isn't being raised. I would delete the return on line 61 so that this can re-run the menu